ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/transaction/UtxoLedgerTransaction.java

```diff
@@ -0,0 +1,296 @@
+     * @param type The type of the {@link Command}.
+     * @return Returns all commands that match the specified type.
+     */
+    @NotNull
+    <T extends Command> List<T> getCommands(@NotNull Class<T> type);
+
+    /**
+     * Gets the input state refs associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the input state refs associated with the current {@link UtxoLedgerTransaction}.
+     */
+    @NotNull
+    List<StateRef> getInputStateRefs();
+
+    /**
+     * Gets the input states and state refs associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the input states and state refs associated with the current {@link UtxoLedgerTransaction}.
+     */
+    @NotNull
+    List<StateAndRef<?>> getInputStateAndRefs();
+
+    /**
+     * Obtains all input states and state refs that match the specified type.
+     *
+     * @param <T>  The underlying type of the {@link ContractState}.
+     * @param type The type of the {@link ContractState}.
+     * @return Returns all input states and state refs that match the specified type.
+     */
+    @NotNull
+    <T extends ContractState> List<StateAndRef<T>> getInputStateAndRefs(@NotNull Class<T> type);
+
+    /**
+     * Gets the input transaction states associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the input transaction states associated with the current {@link UtxoLedgerTransaction}.
+     */
+    @NotNull
+    default List<TransactionState<?>> getInputTransactionStates() {
+        return getInputStateAndRefs()
+                .stream()
+                .map(StateAndRef::getState)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets the input contract states associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the input contract states associated with the current {@link UtxoLedgerTransaction}.
+     */
+    @NotNull
+    default List<ContractState> getInputContractStates() {
+        return getInputTransactionStates()
+                .stream()
+                .map(TransactionState::getContractState)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Obtains all input contract states that match the specified type.
+     *
+     * @param <T>  The underlying type of the {@link ContractState}.
+     * @param type The type of the {@link ContractState}.
+     * @return Returns all input contract states that match the specified type.
+     */
+    // TODO : Rename to getInputContractStates
+    @NotNull
+    <T extends ContractState> List<T> getInputStates(@NotNull Class<T> type);
+
+    /**
+     * Gets the reference input state refs associated with the current {@link UtxoLedgerTransaction}.
+     *
+     * @return Returns the reference input state refs associated with the current {@link UtxoLedgerTransaction}.
+     */
+    // TODO : Rename to getReferenceInputStateRefs
```

Comment:
Regarding adding back the `input` to the name of these functions:
We had a short discussion on slack in December where it seemed that the shorter (without `input`) is the preferred option.